internal/locking_buffer.go

```diff
@@ -0,0 +1,145 @@
+package internal
+
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+// LockingBuffer wraps a bytes.Buffer with a mutex.
+type LockingBuffer struct {
+	mu sync.Mutex
+	b  bytes.Buffer
+}
```

Comment:
The `LockingBuffer` was added to handle reading STDOUT/STDERR while the external `litestream` process was writing to it at the same time.